lib/web_ui/lib/src/engine/browser_detection.dart

```diff
@@ -149,11 +149,23 @@ OperatingSystem get operatingSystem {
 /// This is intended to be used for testing and debugging only.
 OperatingSystem? debugOperatingSystemOverride;
 
-OperatingSystem _detectOperatingSystem() {
-  final String platform = html.window.navigator.platform!;
-  final String userAgent = html.window.navigator.userAgent;
+@visibleForTesting
+OperatingSystem detectOperatingSystem({
+  @visibleForTesting String? overridePlatform,
+  @visibleForTesting String? overrideUserAgent,
+  @visibleForTesting int? overrideMaxTouchPoints,
```

Comment:
You don't need `@visibleForTesting` on parameters

Reply:
Doesn't this enforce *not* using these parameters outside of tests?

Reply:
(It's probably allowed within the library anyway, so for all intents and purposes these don't do anything :P)

Reply:
(Removed)